### DISCUSSION TITLE
motor designs

### DISCUSSION THREAD

**Discussion by EllenGwen** [Eomys/pyleecan#343] [Q&A]
Hi,
I'm new to this and currently exploring with a motor that I am trying to create. Is there a way to introduce rotor skewing ? Also, wondering if there are ways to alter the shape of ventialtion ducts other than the given three ?

**Comment by BonneelP** [comment]
Hello,
Thank you for your interest in pyleecan :)
For now we haven't implemented the skew. We have an open issue to talk about this feature: #25 We have some parameters that can be used to play with the angle in the Magnetic model that can help to introduce the skew.
Regarding vents, you can use any "Hole" object which mean that you could use the Hole available in the SynRM GUI for instance. You can also use the Hole_DXF to define any ventilation shape from a DXF. But these options are not available in the GUI for now, you need to define a script for that (you can create the machine in the GUI, load it in a script and set only the ventilation in the script).
Hope it helps, feel free to ask some further questions if needed :)

Best regards,
Pierre

**Reply by SebGue** [reply]
Hello @BonneelP ,

I think setting an offset to the rotor is currently not possible (see our rotor angle discussion #295). 
Otherwise one could utilize the multisimulation feature to simulate skewing.
Regarding vents, I got to open an issue, since the vents may be on the lamination label in FEMM. So the simulation fails. Further I think there is also the vents on boundaries issue open.

Hello @EllenGwen , 
maybe you could add a sketch of your vent geometry. So we could give some advice on how to setup a new vent, starting with an existing class.

Best regards, Sebastian

**Reply by EllenGwen** [reply]
Thank you both for your replies. The vent is simply oval shapes  instead of a circle done using VentilationCirc.

**Reply by BonneelP** [reply]
Great idea :) If we solve issue #295 then a multisimulation would do the trick.

Regarding lamination reference point, to solve that we need to find a radius that doesn't collide with any hole or vent. For instance by using the "comp_radius" methods to create something like "get_bore_desc" that would return Rint=>R1:lamination, R1=>R2: vent, R2=>R3 lamination, R3=>R4: hole[0], R4=> Rext lamination. Then we can add a clever rule to set (for instance) point_ref = (R2+R3)/2*exp(1j*pi/sym). Then as we had a lamination with "no lamination" (#334 )  we can have laminations without any radius where all point are part of the lamination but we can maybe deal with this second issue later. 

Regarding vent on boundaries, issue #24 will solve that (normally it should already work since we use this method to split winding, we "just" need to properly check it)

Regarding oval, for now we are not able to draw spline so we would need to define the oval as several Arc (it seems doable). But introducing spline would be a great addition to pyleecan, in particular for DXF import.

Best regards,
Pierre

**Reply by SebGue** [reply]
Hello @EllenGwen ,

if you want to setup a new vent shape, you should start with 'pyleecan\Methods\Slot\VentilationCirc\build_geometry' method and the respective Class csv file to see the VentilationCirc properties.
Then you need to define your oval shape parameters and modify a copy of build_geometry respectively. I.e. you may use SurfLine instead of Circle for now.
If you done with that you can setup a new class from this. Give a sign, if you need further help on that.

Hello @BonneelP ,

I think we should also add a minimal example to the website on how to create a new class from an existing one as a guideline for new users.

Best regards, Sebastian
